### PR TITLE
Integration spec and fix n+1

### PR DESCRIPTION
**Pull request by ankitt26**
In this PR, I did -

-  N+1 problem is solved when fetching all posts and their comments for a user.
 
-  screenshots of your app's logs before and after the fix**
   
- before -- 
before n + 1
- After --
after n +1

#### Integration specs
- Use Capybara to write integration tests for each view in your project.
- Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  describe 'User index page' do
+    before(:each) { visit user_path(id: @user.id) }
+    it 'displays the photos and username of user' do
+      expect(page.has_xpath?("//img[@src='https://example.com/photo.jpg']"))
+      expect(page).to have_content('Ankit')
+    end
+    it "shows the number of posts of user, user bio, and the user's first 3 posts " do
+      expect(page).to have_content('Number of posts: 4')
+      expect(page).to have_content('I am a new Gen coder')
+      expect(page).not_to have_content('This is the first post.')
```
Comment (tugsem) [inline spec/features/user_show_spec.rb:23]: - [ ] This test asserts the first post cannot be found on the page. However, as per the requirements, you need to test the first 3 posts should be on the page. Please add it.
image

File: spec/features/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      expect(page).to have_content('This is a comment')
+      expect(page).to have_content('Comments: 1,')
+      expect(page).to have_content('Likes: 1')
+      expect(page).to have_content('1')
+    end
+
+    it 'redirects to post show page when clicking on a post' do
+      click_link 'Post 1'
+      expect(page).to have_current_path(user_post_path(@user, @post1))
+    end
```
Comment (tugsem) [inline spec/features/post_index_spec.rb:54]: - [ ] As per the requirement, please add one more test for asserts there is a pagination button on the page.
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @ankitt26  and @iamsjunaid 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️the app tested with capybara gem
✔️n+1 query problem fixed
✔️professional documentation
## Required Changes ♻️
- [ ] Please add package.json file into your directory, when I run 'npm i' it throws an error. 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks, @tugsem for requesting the changes, I will make the changes to the code 🙋🏿‍♂️.

**Review by levy002 — APPROVED** [review]
## Hello @ankitt26 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter error :heavy_check_mark: 
- You implemented all suggested changes :heavy_check_mark: 

## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**